pkg/client/tools.go

```diff
@@ -39,20 +40,42 @@ import (
 // ImageImportIntoClusterMulti starts up a k3d tools container for the selected cluster and uses it to export
 // images from the runtime to import them into the nodes of the selected cluster
 func ImageImportIntoClusterMulti(ctx context.Context, runtime runtimes.Runtime, images []string, cluster *k3d.Cluster, opts k3d.ImageImportOpts) error {
+
+	// stdin case
+	if len(images) == 1 && images[0] == "-" {
+		loadImageFromStream(ctx, runtime, os.Stdin, cluster)
+		return nil
+	}
+
 	imagesFromRuntime, imagesFromTar, err := findImages(ctx, runtime, images)
 	if err != nil {
 		return fmt.Errorf("failed to find images: %w", err)
 	}
 
 	// no images found to load -> exit early
 	if len(imagesFromRuntime)+len(imagesFromTar) == 0 {
-		return fmt.Errorf("No valid images specified")
+		return fmt.Errorf("no valid images specified")
 	}
 
-	// create tools node to export images
-	toolsNode, err := EnsureToolsNode(ctx, runtime, cluster)
-	if err != nil {
-		return fmt.Errorf("failed to ensure that tools node is running: %w", err)
+	loadWithToolsNode := false
+
+	switch opts.Mode {
+	case k3d.ImportModeAutoDetect:
+		if err != nil {
+			return fmt.Errorf("failed to retrieve container runtime information: %w", err)
+		}
+		if err != nil {
+			return fmt.Errorf("failed to compile remote runtime endpoint regexp: %w", err)
+		}
```

Comment:
Uhm.. what happened here? :thinking:

Reply:
oops, must have missed something after refactoring here :D i initially thought i have to parse the endpoint string but the runtime already provides the address